servers/express.js

```diff
@@ -65,51 +84,51 @@ const eventloopDataRows = [];
 const memoryDataRows = [];
 const cpuDataRows = [];
 
-let lastNotification = Date.now();
 async function collector() {
   const lines = watcher(pathToLogFile);
   const first = await lines.next();
-  // maybe wait for the file to appear? idk.
+
   if (first.done) {
-    throw new Error('no log lines to read');
+    throw new Error('No log lines to read');
   }
-  const firstRecord = JSON.parse(first.value);
-  const firstTs = firstRecord.ts;
+  const firstTs = JSON.parse(first.value).ts;
+  
   for await (const line of lines) {
-    if (line === null) {
-      continue;
-    }
-    // add to data rows - rebase time off of first record ts
-    // look for type === eventloop
-    // set eventloopDataRows to [deltaTime, 50, 75, 90, 95, 99] values
+    if (line === null) continue;
     try {
       const record = JSON.parse(line);
-      if (record.type !== 'eventloop') {
-        continue;
-      }
-      const delta = (record.ts - firstTs) / 1000;
       const entry = record.entry;
-      // eventloop delay is in nanoseconds; make them ms.
-      const row = [delta];
-      const percentiles = [50, 75, 90, 95, 99];
-      for (let i = percentiles.length - 1; i >= 0 ; i--) {
-        row.push(entry[percentiles[i]] / 1000000);
-      }
-      eventloopDataRows.push(`[${row}]`);
-      if (Date.now() - lastNotification > 60 * 1000) {
-        lastNotification = Date.now();
-        console.log(`[total data rows: ${eventloopDataRows.length}]`);
-      }
+      const delta = (record.ts - firstTs) / 1e3;
+      
+      if (record.type === 'eventloop') {
+        // Eventloop delay is in nanoseconds. Make it ms.
+        const row = [delta];
+        const percentiles = [50, 75, 90, 95, 99];
+        for (let i = percentiles.length - 1; i >= 0 ; i--) {
+          row.push(entry[percentiles[i]] / 1e6);
+        }
+        eventloopDataRows.push(`[${row}]`);
+      } else if (record.type === 'proc') {
+        // Memory data is in bytes. Make it megabytes
+        const externalAvg = entry['externalAvg'] / 1e6;
+        const heapUsedAvg = entry['heapUsedAvg'] / 1e6;
+        const heapTotal = entry['heapTotal'] / 1e6;
+        const rss = entry['rss'] / 1e6;
+
+        // Cpu data is in microseconds. Make it ms
+        cpuDataRows.push(`[${delta}, ${entry['cpuUserAvg'] / 1e3}, ${entry['cpuSystemAvg'] / 1e3}]`);
+        memoryDataRows.push(`[${delta}, ${rss}, ${heapTotal}, ${heapUsedAvg}, ${externalAvg}]`);
+      } else continue;
```

Comment:
this i definitely don't like - to me `else` should always follow the form of the previous if/else-if blocks for consistency, so

```` 
} else {
     continue;
}
```